### PR TITLE
fix: casper support on memo tag send flow

### PR DESCRIPTION
**Pull request by themooneer**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

This work consist to support Casper send flow within the new Memo/Tag flow. 
- add MemoTag (TransferId)  field in the first step of the send flow (keeping the same label/wording)
- on feature flag off, the legacy input will be shown on the second step.

#### **_NOTE before the review  (please read this)_**  🚨
- business logic in the apps/ledger-live-desktop/src/renderer/families/casper/MemoField.tsx component is a copy for apps/ledger-live-desktop/src/renderer/families/casper/TransferId.tsx component with just an input change  in order to secure the legacy with the feature flag. -> it will be deleted on the deletion of lldMemoTag

##### Feature flag 📴 Preview 

https://github.com/user-attachments/assets/520ee418-8ee5-48eb-8bca-7d3f3c9a87a1

##### Feature flag 🔛 Preview

https://github.com/user-attachments/assets/8bf0b4c2-96db-4375-8d6c-abda3613fc15

### ❓ Context

- **JIRA or GitHub link**: 

https://ledgerhq.atlassian.net/browse/LIVE-14921

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/features/MemoTag/utils.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { MEMO_TAG_COINS } from "./constants";
  * @returns {string | undefined} The memo tag value if the transaction family is supported, otherwise undefined.
  */
 export const getMemoTagValueByTransactionFamily = (transaction: Transaction) => {
+  console.log(transaction);
```
Comment (thesan) [inline apps/ledger-live-desktop/src/newArch/features/MemoTag/utils.ts:13]: I think you forgot this one

File: apps/ledger-live-desktop/src/newArch/features/MemoTag/__tests__/MemoTagField.test.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import React from "react";
 import { render, screen, fireEvent } from "tests/testUtils";
 import MemoTagField from "../components/MemoTagField";
+import { act } from "react-dom/test-utils";
```
Comment (thesan) [inline apps/ledger-live-desktop/src/newArch/features/MemoTag/__tests__/MemoTagField.test.tsx:8]: Is this used ?

File: apps/ledger-live-desktop/src/newArch/features/MemoTag/components/MemoTagField.tsx [hunk 3]
```diff
@@ -38,28 +41,33 @@ const MemoTagField = ({
   maxMemoLength,
   CaracterCountComponent,
   autoFocus,
+  placeholder,
+  label,
+  tooltipText,
 }: MemoTagFieldProps) => {
```
Comment (thesan) [inline apps/ledger-live-desktop/src/newArch/features/MemoTag/components/MemoTagField.tsx:47]: It's a small detailed but MemoTagFieldProps shares the same name with MemoTagFieldProps from `apps/ledger-live-desktop/src/renderer/families/casper/types.ts`. Maybe just calling the type `Props` would make things clearer

File: apps/ledger-live-desktop/src/newArch/features/MemoTag/utils.ts (hunk 1 above)
Comment (themooneer) [reply apps/ledger-live-desktop/src/newArch/features/MemoTag/utils.ts]: ooops ! fixed

File: apps/ledger-live-desktop/src/newArch/features/MemoTag/__tests__/MemoTagField.test.tsx (hunk 2 above)
Comment (themooneer) [reply apps/ledger-live-desktop/src/newArch/features/MemoTag/__tests__/MemoTagField.test.tsx]: done

**Review by thesan — APPROVED** [review]

**Review by Wozacosta — DISMISSED** [review]

**Review by Wozacosta — APPROVED** [review]

**Review by thesan — APPROVED** [review]

**Pull request merged**